<a href="https://colab.research.google.com/github/Pranaybannu/PAISABAZAR-BANKING-FRAUD-ANALYSIS/blob/main/BANKING_FRAUD_PAISABAZAR.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Project Name**    -



##### **Project Type**    - PAISABAZAR BANKING FRAUD ANALYSIS(EDA)
##### **Contribution**    - Individual(PRANAY RAJ SAMBODHU)

# **Project Summary -**

Assessing the credit score of an individual is crucial for banking and financial institutions, as credit score serves as a key metric to evaluate the likelihood of loan repayment. Banks and lenders rely on credit scores to identify potential customers who are more likely to repay their approved loans, thereby minimizing financial risk and supporting informed lending decisions.

This project involves analyzing customer financial behavior along with various credit-related attributes to understand the factors that influence an individual’s credit score. The dataset includes customer demographic information such as age, name, SSN, and occupation, as well as financial details like annual and monthly income, number and types of loans, number of bank accounts, number of credit cards, delayed payment counts and duration, credit mix, outstanding debt, and credit utilization ratio. These variables collectively represent a customer’s financial profile and will be used for exploratory data analysis (EDA) and visualization.

The primary objective of analyzing customer financial information is to address the following key questions:

1. Does a higher annual income correlate with a higher credit score?

2. Does the credit utilization ratio have a greater impact on credit scores compared to the number of loans taken?

3. Is having a higher number of credit cards associated with a better credit score, assuming the credit utilization ratio is well managed?

4. What is the relationship between delayed payments and credit score categories?

The exploratory data analysis process will involve the following steps:

1. Loading the dataset and identifying its structure, including the number of rows and columns.

2. Examining the data types of each column to determine their suitability for analysis.

3. Performing data cleaning by handling duplicate records, missing values, special characters, and other inconsistencies.

4. Conducting data wrangling, including data type conversion, feature engineering, and creation of new variables where required.

5. Visualizing data using libraries such as Matplotlib, Seaborn, and Plotly to identify patterns and trends.

6. Extracting meaningful insights from the visualizations, highlighting key positives and negatives from a business perspective.

To enhance accessibility and usability, the final insights will be consolidated into an interactive Streamlit dashboard, enabling users to dynamically explore trends, relationships, and key indicators related to credit score assessment.

# **GitHub Link -**

Provide your GitHub Link here.

# **Problem Statement**


1. Does a **higher annual income correlate with a higher credit score**?

2. Does the **credit utilization ratio** have a greater **impact on credit scores** compared to the number of loans taken?

3. Is having a **higher number of credit cards associated with a better credit score,** assuming the credit utilization ratio is well managed?

4. What is the **relationship between delayed payments and credit score** categories?

#### **Define Your Business Objective?**

1. Identifyng the **High risk customers**,**loyal customers** and **High-valued customers**.

2. Using the **credit scroes,Incomes** and **credit mix** as **key metrics** to **assess customers's financial behaviour.**

# ***Let's Begin !***

## ***1. Know Your Data***

### Import Libraries

In [ ]:
# Import Libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import plotly.express as px
import plotly.graph_objects as go


### Dataset Loading

In [ ]:
# Load Dataset
from google.colab import drive
drive.mount('/content/drive')


Mounted at /content/drive


In [ ]:
pdf = pd.read_csv('/content/drive/MyDrive/paisabazar/dataset.csv')

### Dataset First View

In [ ]:
pd.set_option('display.max_rows',None)
pd.set_option('display.max_columns',None)   ## to display all rows and columns

In [ ]:
# Dataset First Look
pdf.head(10)

,ID,Customer_ID,Month,Name,Age,SSN,Occupation,Annual_Income,Monthly_Inhand_Salary,Num_Bank_Accounts,Num_Credit_Card,Interest_Rate,Num_of_Loan,Type_of_Loan,Delay_from_due_date,Num_of_Delayed_Payment,Changed_Credit_Limit,Num_Credit_Inquiries,Credit_Mix,Outstanding_Debt,Credit_Utilization_Ratio,Credit_History_Age,Payment_of_Min_Amount,Total_EMI_per_month,Amount_invested_monthly,Payment_Behaviour,Monthly_Balance,Credit_Score
0,5634,3392,1,Aaron Maashoh,23.0,821000265.0,Scientist,19114.12,1824.843333,3.0,4.0,3.0,4.0,"Auto Loan, Credit-Builder Loan, Personal Loan,...",3.0,7.0,11.27,4.0,Good,809.98,26.822620,265.0,No,49.574949,21.465380,High_spent_Small_value_payments,312.494089,Good
1,5635,3392,2,Aaron Maashoh,23.0,821000265.0,Scientist,19114.12,1824.843333,3.0,4.0,3.0,4.0,"Auto Loan, Credit-Builder Loan, Personal Loan,...",3.0,4.0,11.27,4.0,Good,809.98,31.944960,266.0,No,49.574949,21.465380,Low_spent_Large_value_payments,284.629162,Good
2,5636,3392,3,Aaron Maashoh,23.0,821000265.0,Scientist,19114.12,1824.843333,3.0,4.0,3.0,4.0,"Auto Loan, Credit-Builder Loan, Personal Loan,...",3.0,7.0,11.27,4.0,Good,809.98,28.609352,267.0,No,49.574949,21.465380,Low_spent_Medium_value_payments,331.209863,Good
3,5637,3392,4,Aaron Maashoh,23.0,821000265.0,Scientist,19114.12,1824.843333,3.0,4.0,3.0,4.0,"Auto Loan, Credit-Builder Loan, Personal Loan,...",5.0,4.0,6.27,4.0,Good,809.98,31.377862,268.0,No,49.574949,21.465380,Low_spent_Small_value_payments,223.451310,Good
4,5638,3392,5,Aaron Maashoh,23.0,821000265.0,Scientist,19114.12,1824.843333,3.0,4.0,3.0,4.0,"Auto Loan, Credit-Builder Loan, Personal Loan,...",6.0,4.0,11.27,4.0,Good,809.98,24.797347,269.0,No,49.574949,21.465380,High_spent_Medium_value_payments,341.489231,Good
5,5639,3392,6,Aaron Maashoh,23.0,821000265.0,Scientist,19114.12,1824.843333,3.0,4.0,3.0,4.0,"Auto Loan, Credit-Builder Loan, Personal Loan,...",8.0,4.0,9.27,4.0,Good,809.98,27.262259,270.0,No,49.574949,21.465380,High_spent_Medium_value_payments,340.479212,Good
6,5640,3392,7,Aaron Maashoh,23.0,821000265.0,Scientist,19114.12,1824.843333,3.0,4.0,3.0,4.0,"Auto Loan, Credit-Builder Loan, Personal Loan,...",3.0,8.0,11.27,4.0,Good,809.98,22.537593,271.0,No,49.574949,21.465380,Low_spent_Small_value_payments,244.565317,Good
7,5641,3392,8,Aaron Maashoh,23.0,821000265.0,Scientist,19114.12,1824.843333,3.0,4.0,3.0,4.0,"Auto Loan, Credit-Builder Loan, Personal Loan,...",3.0,6.0,11.27,4.0,Good,809.98,23.933795,272.0,No,49.574949,21.465380,High_spent_Medium_value_payments,358.124168,Standard
8,5646,8625,1,Rick Rothackerj,28.0,4075839.0,Teacher,34847.84,3037.986667,2.0,4.0,6.0,1.0,Credit-Builder Loan,3.0,4.0,5.42,2.0,Good,605.03,24.464031,319.0,No,18.816215,39.684018,Low_spent_Small_value_payments,470.690627,Standard
9,5647,8625,2,Rick Rothackerj,28.0,4075839.0,Teacher,34847.84,3037.986667,2.0,4.0,6.0,1.0,Credit-Builder Loan,7.0,1.0,7.42,2.0,Good,605.03,38.550848,320.0,No,18.816215,39.684018,High_spent_Large_value_payments,484.591214,Good


### Dataset Rows & Columns count

In [ ]:
# Dataset Rows & Columns count
pdf.shape ## has 100krows and 28 columns

(100000, 28)

### Dataset Information

In [ ]:
# Dataset Info
pdf.info()  ## dataset contains different datatypes and has no missing or null values

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 100000 entries, 0 to 99999
Data columns (total 28 columns):
 #   Column                    Non-Null Count   Dtype  
---  ------                    --------------   -----  
 0   ID                        100000 non-null  int64  
 1   Customer_ID               100000 non-null  int64  
 2   Month                     100000 non-null  int64  
 3   Name                      100000 non-null  object 
 4   Age                       100000 non-null  float64
 5   SSN                       100000 non-null  float64
 6   Occupation                100000 non-null  object 
 7   Annual_Income             100000 non-null  float64
 8   Monthly_Inhand_Salary     100000 non-null  float64
 9   Num_Bank_Accounts         100000 non-null  float64
 10  Num_Credit_Card           100000 non-null  float64
 11  Interest_Rate             100000 non-null  float64
 12  Num_of_Loan               100000 non-null  float64
 13  Type_of_Loan              100000 non-null  ob

#### Duplicate Values

In [ ]:
# Dataset Duplicate Value Count
pdf.duplicated().value_counts()           ##there are no duplicate values in the dataset.

,count
False,100000


#### Missing Values/Null Values

In [ ]:
# Missing Values/Null Values Count
pdf.isnull().mean()*100  ##no null values or inconsistencies in the data.

,0
ID,0.0
Customer_ID,0.0
Month,0.0
Name,0.0
Age,0.0
SSN,0.0
Occupation,0.0
Annual_Income,0.0
Monthly_Inhand_Salary,0.0
Num_Bank_Accounts,0.0


### What did you know about your dataset?

Dataset is **cleaned,** it has **100k rows and 28 columns**.There are **no null** values and **duplicated rows/columns** in the dataset. The dataset contains columns with **multiple datatypes**.

## ***2. Understanding Your Variables***

In [ ]:
# Dataset Columns
pdf.columns

Index(['ID', 'Customer_ID', 'Month', 'Name', 'Age', 'SSN', 'Occupation',
       'Annual_Income', 'Monthly_Inhand_Salary', 'Num_Bank_Accounts',
       'Num_Credit_Card', 'Interest_Rate', 'Num_of_Loan', 'Type_of_Loan',
       'Delay_from_due_date', 'Num_of_Delayed_Payment', 'Changed_Credit_Limit',
       'Num_Credit_Inquiries', 'Credit_Mix', 'Outstanding_Debt',
       'Credit_Utilization_Ratio', 'Credit_History_Age',
       'Payment_of_Min_Amount', 'Total_EMI_per_month',
       'Amount_invested_monthly', 'Payment_Behaviour', 'Monthly_Balance',
       'Credit_Score'],
      dtype='object')

In [ ]:
# Dataset Describe
pdf.describe(include='number')  ##for int,float columns

,ID,Customer_ID,Month,Age,SSN,Annual_Income,Monthly_Inhand_Salary,Num_Bank_Accounts,Num_Credit_Card,Interest_Rate,Num_of_Loan,Delay_from_due_date,Num_of_Delayed_Payment,Changed_Credit_Limit,Num_Credit_Inquiries,Outstanding_Debt,Credit_Utilization_Ratio,Credit_History_Age,Total_EMI_per_month,Amount_invested_monthly,Monthly_Balance
count,100000.000000,100000.000000,100000.000000,100000.000000,1.000000e+05,100000.000000,100000.000000,100000.000000,100000.000000,100000.00000,100000.000000,100000.00000,100000.000000,100000.000000,100000.000000,100000.000000,100000.000000,100000.000000,100000.000000,100000.000000,100000.000000
mean,80631.500000,25982.666640,4.500000,33.316340,5.004617e+08,50505.123449,4197.270835,5.368820,5.533570,14.53208,3.532880,21.08141,13.313120,10.470323,5.798250,1426.220376,32.285173,221.220460,107.699208,55.101315,392.697586
std,43301.486619,14340.543051,2.291299,10.764812,2.908267e+08,38299.422093,3186.432497,2.593314,2.067098,8.74133,2.446356,14.80456,6.237166,6.609481,3.867826,1155.129026,5.116875,99.680716,132.267056,39.006932,201.652719
min,5634.000000,1006.000000,1.000000,14.000000,8.134900e+04,7005.930000,303.645417,0.000000,0.000000,1.00000,0.000000,0.00000,0.000000,0.500000,0.000000,0.230000,20.000000,1.000000,0.000000,0.000000,0.007760
25%,43132.750000,13664.500000,2.750000,24.000000,2.451686e+08,19342.972500,1626.594167,3.000000,4.000000,7.00000,2.000000,10.00000,9.000000,5.380000,3.000000,566.072500,28.052567,144.000000,29.268886,27.959111,267.615983
50%,80631.500000,25777.000000,4.500000,33.000000,5.006886e+08,36999.705000,3095.905000,5.000000,5.000000,13.00000,3.000000,18.00000,14.000000,9.400000,5.000000,1166.155000,32.305784,219.000000,66.462304,45.156550,333.865366
75%,118130.250000,38385.000000,6.250000,42.000000,7.560027e+08,71683.470000,5957.715000,7.000000,7.000000,20.00000,5.000000,28.00000,18.000000,14.850000,8.000000,1945.962500,36.496663,302.000000,147.392573,71.295797,463.215683
max,155629.000000,50999.000000,8.000000,56.000000,9.999934e+08,179987.280000,15204.633333,11.000000,11.000000,34.00000,9.000000,62.00000,25.000000,29.980000,17.000000,4998.070000,50.000000,404.000000,1779.103254,434.191089,1183.930696


In [ ]:
##describe for object columns
pdf.describe(include='object')

,Name,Occupation,Type_of_Loan,Credit_Mix,Payment_of_Min_Amount,Payment_Behaviour,Credit_Score
count,100000,100000,100000,100000,100000,100000,100000
unique,10128,15,6261,3,3,6,3
top,Jessicad,Lawyer,No Data,Standard,Yes,Low_spent_Small_value_payments,Standard
freq,48,7096,11408,45848,52326,28616,53174


### Variables Description

**NUMERICAL TYPE COLUMNS**

1. **IDS**: 3392(int)
2. **Month**: 1/2/3(int)  
3. **Age:** 23.0(float)
4. **SSN**: 821000265.0(float)
5. **Annual_income:** 19114.12(float)
6. **Monthly_Inhand_Salary**:1824.84333(float)
7. **Num_Bank_Accounts**: 3.0(float)
8. **Num_Credit_Card**:4.0 (float)
9. **Interest_Rate**:3.0 (float)
10. **Num_of_Loan**:4.0	(float)
11. **Delay_from_due_date**:3.0	(float)
12. **Num_of_Delayed_Payment**:7.0(float)
13. **Changed_Credit_Limit**: 11.27(float)
14. **Num_Credit_Inquiries**: 4.0(float)
15.  **Outstanding_Debt**: 809.98(float)
16. **Credit_Utilization_Ratio:** 26.8336(float)
17. **Credit_History_Age**: 265.0(float)
18. **Total_EMI_per_month:** 49.574949(float)
19. **Amount_invested_monthly**:21.465380(float)
20. **Monthly_Balance:** 284.629162(float)

**OBJECT TYPE COLUMNS:**
1. **Name**: Aaron mashoh(string)
2. **Occupation**: scientist(string)
3. **Type_of_Loan**: Auto/credit-builder(string in list)
4. **Credit_Mix**: Good/bad(string)
5.** Payment_of_Min_Amount**: yes/no(string)
6. **Payment_Behaviour**: High_spent_Small_value_payments(string)
7. **Credit_Score**: Good/bad/saisfactory(string)



### Check Unique Values for each variable.

In [ ]:
pdf.select_dtypes(include='object').columns

Index(['Name', 'Occupation', 'Type_of_Loan', 'Credit_Mix',
       'Payment_of_Min_Amount', 'Payment_Behaviour', 'Credit_Score'],
      dtype='object')

In [ ]:
# Check Unique Values for each variable.
pdf['Occupation'].value_counts()  ## most of the customers are lawyers,engineers,architects

,count
Occupation,
Lawyer,7096
Engineer,6864
Architect,6824
Mechanic,6776
Accountant,6744
Scientist,6744
Media_Manager,6720
Developer,6720
Teacher,6672


In [ ]:
pdf['Age'].value_counts()  ##mid 20s to high 30s are more

,count
Age,
38.0,3070
28.0,3045
31.0,3037
26.0,3025
32.0,2969
36.0,2953
25.0,2952
27.0,2951
35.0,2940


In [ ]:
pdf['Type_of_Loan'].value_counts().head()  #credit builder loan,personal loan are the top loan types

,count
Type_of_Loan,
No Data,11408
Not Specified,1408
Credit-Builder Loan,1280
Personal Loan,1272
Debt Consolidation Loan,1264


In [ ]:
pdf['Credit_Mix'].value_counts()  ##standard is the top credit-mix

,count
Credit_Mix,
Standard,45848
Good,30384
Bad,23768


In [ ]:
pdf['Payment_of_Min_Amount'].value_counts() ## most customers are paying more than minimum amount

,count
Payment_of_Min_Amount,
Yes,52326
No,35667
NM,12007


In [ ]:
pdf['Payment_Behaviour'].value_counts()   ##Low_spent_Small_value_payments is top behaviour among the cutsomers

,count
Payment_Behaviour,
Low_spent_Small_value_payments,28616
High_spent_Medium_value_payments,19738
High_spent_Large_value_payments,14726
Low_spent_Medium_value_payments,14399
High_spent_Small_value_payments,11764
Low_spent_Large_value_payments,10757


In [ ]:
pdf['Credit_Score'].value_counts()  ##standard credit score is maintainded by most customers

,count
Credit_Score,
Standard,53174
Poor,28998
Good,17828


## 3. ***Data Wrangling***

### Data Wrangling Code

In [ ]:
# Write your code to make your dataset analysis ready.
pdf['Type_of_Loan'].value_counts().tail()   ##this column contains coma seperaed strings

##making a new column which will contain only the primay loan type. This is necessary for visualization

,count
Type_of_Loan,
"Student Loan, Auto Loan, Student Loan, Credit-Builder Loan, Home Equity Loan, Debt Consolidation Loan, and Debt Consolidation Loan",8
"Debt Consolidation Loan, Personal Loan, Mortgage Loan, Personal Loan, Not Specified, Mortgage Loan, and Home Equity Loan",8
"Student Loan, Home Equity Loan, Student Loan, Personal Loan, Not Specified, Auto Loan, Auto Loan, and Debt Consolidation Loan",8
"Not Specified, Auto Loan, Auto Loan, and Debt Consolidation Loan",8
"Payday Loan, Mortgage Loan, Home Equity Loan, Home Equity Loan, Auto Loan, Not Specified, Credit-Builder Loan, and Home Equity Loan",8


In [ ]:
pdf['primary_loan_type'] = pdf['Type_of_Loan'].str.split(',')

In [ ]:
pdf['primary_loan_type'] = pdf['primary_loan_type'].str[0]

In [ ]:
pdf['primary_loan_type'].value_counts().tail()

,count
primary_loan_type,
Student Loan,9672
Debt Consolidation Loan,9648
Mortgage Loan,9632
Auto Loan,9568
Home Equity Loan,9360


In [ ]:
pdf.head()

,ID,Customer_ID,Month,Name,Age,SSN,Occupation,Annual_Income,Monthly_Inhand_Salary,Num_Bank_Accounts,Num_Credit_Card,Interest_Rate,Num_of_Loan,Type_of_Loan,Delay_from_due_date,Num_of_Delayed_Payment,Changed_Credit_Limit,Num_Credit_Inquiries,Credit_Mix,Outstanding_Debt,Credit_Utilization_Ratio,Credit_History_Age,Payment_of_Min_Amount,Total_EMI_per_month,Amount_invested_monthly,Payment_Behaviour,Monthly_Balance,Credit_Score,primary_loan_type
0,5634,3392,1,Aaron Maashoh,23.0,821000265.0,Scientist,19114.12,1824.843333,3.0,4.0,3.0,4.0,"Auto Loan, Credit-Builder Loan, Personal Loan,...",3.0,7.0,11.27,4.0,Good,809.98,26.822620,265.0,No,49.574949,21.46538,High_spent_Small_value_payments,312.494089,Good,Auto Loan
1,5635,3392,2,Aaron Maashoh,23.0,821000265.0,Scientist,19114.12,1824.843333,3.0,4.0,3.0,4.0,"Auto Loan, Credit-Builder Loan, Personal Loan,...",3.0,4.0,11.27,4.0,Good,809.98,31.944960,266.0,No,49.574949,21.46538,Low_spent_Large_value_payments,284.629162,Good,Auto Loan
2,5636,3392,3,Aaron Maashoh,23.0,821000265.0,Scientist,19114.12,1824.843333,3.0,4.0,3.0,4.0,"Auto Loan, Credit-Builder Loan, Personal Loan,...",3.0,7.0,11.27,4.0,Good,809.98,28.609352,267.0,No,49.574949,21.46538,Low_spent_Medium_value_payments,331.209863,Good,Auto Loan
3,5637,3392,4,Aaron Maashoh,23.0,821000265.0,Scientist,19114.12,1824.843333,3.0,4.0,3.0,4.0,"Auto Loan, Credit-Builder Loan, Personal Loan,...",5.0,4.0,6.27,4.0,Good,809.98,31.377862,268.0,No,49.574949,21.46538,Low_spent_Small_value_payments,223.451310,Good,Auto Loan
4,5638,3392,5,Aaron Maashoh,23.0,821000265.0,Scientist,19114.12,1824.843333,3.0,4.0,3.0,4.0,"Auto Loan, Credit-Builder Loan, Personal Loan,...",6.0,4.0,11.27,4.0,Good,809.98,24.797347,269.0,No,49.574949,21.46538,High_spent_Medium_value_payments,341.489231,Good,Auto Loan


In [ ]:
## creating a new column(numercial) for credit score which could be used for correlation heatmap
def crdtmap(x):
  if x=='Poor':
    return 0
  elif x=='Standard':
    return 1
  elif x=='Good':
    return 2

pdf['credit_cat'] = pdf['Credit_Score'].apply(crdtmap)

In [ ]:
pdf.head()

,ID,Customer_ID,Month,Name,Age,SSN,Occupation,Annual_Income,Monthly_Inhand_Salary,Num_Bank_Accounts,Num_Credit_Card,Interest_Rate,Num_of_Loan,Type_of_Loan,Delay_from_due_date,Num_of_Delayed_Payment,Changed_Credit_Limit,Num_Credit_Inquiries,Credit_Mix,Outstanding_Debt,Credit_Utilization_Ratio,Credit_History_Age,Payment_of_Min_Amount,Total_EMI_per_month,Amount_invested_monthly,Payment_Behaviour,Monthly_Balance,Credit_Score,primary_loan_type,credit_cat
0,5634,3392,1,Aaron Maashoh,23.0,821000265.0,Scientist,19114.12,1824.843333,3.0,4.0,3.0,4.0,"Auto Loan, Credit-Builder Loan, Personal Loan,...",3.0,7.0,11.27,4.0,Good,809.98,26.822620,265.0,No,49.574949,21.46538,High_spent_Small_value_payments,312.494089,Good,Auto Loan,2
1,5635,3392,2,Aaron Maashoh,23.0,821000265.0,Scientist,19114.12,1824.843333,3.0,4.0,3.0,4.0,"Auto Loan, Credit-Builder Loan, Personal Loan,...",3.0,4.0,11.27,4.0,Good,809.98,31.944960,266.0,No,49.574949,21.46538,Low_spent_Large_value_payments,284.629162,Good,Auto Loan,2
2,5636,3392,3,Aaron Maashoh,23.0,821000265.0,Scientist,19114.12,1824.843333,3.0,4.0,3.0,4.0,"Auto Loan, Credit-Builder Loan, Personal Loan,...",3.0,7.0,11.27,4.0,Good,809.98,28.609352,267.0,No,49.574949,21.46538,Low_spent_Medium_value_payments,331.209863,Good,Auto Loan,2
3,5637,3392,4,Aaron Maashoh,23.0,821000265.0,Scientist,19114.12,1824.843333,3.0,4.0,3.0,4.0,"Auto Loan, Credit-Builder Loan, Personal Loan,...",5.0,4.0,6.27,4.0,Good,809.98,31.377862,268.0,No,49.574949,21.46538,Low_spent_Small_value_payments,223.451310,Good,Auto Loan,2
4,5638,3392,5,Aaron Maashoh,23.0,821000265.0,Scientist,19114.12,1824.843333,3.0,4.0,3.0,4.0,"Auto Loan, Credit-Builder Loan, Personal Loan,...",6.0,4.0,11.27,4.0,Good,809.98,24.797347,269.0,No,49.574949,21.46538,High_spent_Medium_value_payments,341.489231,Good,Auto Loan,2


In [ ]:
pdf['credit_cat'].info()

<class 'pandas.core.series.Series'>
RangeIndex: 100000 entries, 0 to 99999
Series name: credit_cat
Non-Null Count   Dtype
--------------   -----
100000 non-null  int64
dtypes: int64(1)
memory usage: 781.4 KB


In [ ]:
pdf.to_csv('pdf.csv',index=False)

In [ ]:
from google.colab import files
files.download('pdf.csv')

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

### What all manipulations have you done and insights you found?

I have only done two manipulations for now on the dataset. I have created two new columns **'credit_cat'** and **'primary_loan_type'** which would be **important for anlysis and visualization**. During visualization other manipulations will be done to fit the data into the charts/graphs.

## ***4. Data Vizualization, Storytelling & Experimenting with charts : Understand the relationships between variables***

#### Chart - 1 UNIVARIATE ANLYSIS FOR CREDIT SCORE

In [ ]:
data = pdf['Credit_Score'].value_counts().reset_index()

In [ ]:
data

,Credit_Score,count
0,Standard,53174
1,Poor,28998
2,Good,17828


In [ ]:
# Chart - 1 BAR CHART
fig = px.bar(
    data,
    x='Credit_Score',
    y='count',
    text_auto=False,
    title='Credit score category count using Barchart',
    color_discrete_sequence=px.colors.qualitative.Bold,
    opacity=1
)

fig.update_layout(
    title_font_size=20,
    title_font_weight='bold',
    font= dict(size=12,weight='bold'),
    bargap=0.2
)

fig.show()

##### 1. Why did you pick the specific chart?

Bar chart is used to compare catogeries. Here in this chart we are **credit score categories**

##### 2. What is/are the insight(s) found from the chart?

1. **Standard** credit score contains **highest no.of customers(53.2k**). **Poor** credit score is maintained by **(28.9k) customers** and **Good** credit score is maintained by **lowest no.of people(17.83k)**.

##### 3. Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

**POSITIVES:**
1. There is a chance to **move large segment of customers from Standard credit score to Good credit score** by **promoting good credit behaviour actions** to the customers with standard credit score.

**NEGATIVES:**
1. **Majority** of the customers are in the **medium to high risk category.** There are **very few customers in the low-risk category**.

#### Chart - 2 BIVARIATE ANALYSIS FOR OCCUPATION

Since **occupation and annual income remain constant** in monthly customer records, therefore **customer level aggregation will be done to avoid redundant** data and to ensure accurate box plot visualization.

In [ ]:
data = pdf.groupby('Customer_ID').agg({
    'Occupation':'first',
    'Annual_Income':'first',
    'Credit_Score':'last',                 ###using most recent credit score
    'primary_loan_type':'first'}

).reset_index()

In [ ]:
data.head()

,Customer_ID,Occupation,Annual_Income,Credit_Score,primary_loan_type
0,1006,Journalist,16756.18,Poor,Credit-Builder Loan
1,1007,Manager,21212.91,Poor,Home Equity Loan
2,1008,Developer,33540.43,Standard,No Data
3,1009,Accountant,80983.64,Standard,Credit-Builder Loan
4,1011,Writer,104142.56,Standard,Personal Loan


In [ ]:
data.duplicated().value_counts()   ##no redundant rows or duplicates.

,count
False,12500


In [ ]:
# Chart - 2 Box chart

fig = px.box(
    data,
    x='Credit_Score',
    y='Annual_Income',
    color='Credit_Score',
    title='Box plot for Credit score vs Annual Income'
)

fig.update_layout(
    title_font_size=20,
    title_font_weight = 'bold',
    font= dict(size=12,weight='bold'),
    yaxis_title='Annual income'
)

fig.show()


fig = px.box(
    data,
    x='Occupation',
    y='Annual_Income',
    color='Credit_Score',
    title='Box plot for Occupation vs Annual Income  wrt credit score'
)

fig.update_layout(
    title_font_size=20,
    title_font_weight = 'bold',
    font= dict(size=12,weight='bold'),
    yaxis_title='Incomes'
)

fig.show()


##### 1. Why did you pick the specific chart?

**Box plo**t is used to see the **distribution and potential outliers** of **annual-income** wrt **occupation and credit score**.

##### 2. What is/are the insight(s) found from the chart?

1. Customers with **higher medaian annual income** have** good credit score **.**Poor credit scores** have **lower medain annual income**.

2. Accross **all occupations**, **customers with good credit score have higher median annual income**, compared to the standard and poor categories.

3. As the **credit score improoves,the spread of annual income widens**, indicating **better earnings and income stability**.

##### 3. Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

**POSITIVES:**
1. **Better annual income** can lead to **better credit score**.

#### Chart - 3 BIVARIATE ANALYSIS Histplot for loan type  and income

In [ ]:
# Chart - 3 Histogram
data =pdf.groupby('primary_loan_type')['Annual_Income'].mean().reset_index()


In [ ]:
data.head()

,primary_loan_type,Annual_Income
0,Auto Loan,49095.786334
1,Credit-Builder Loan,47240.216874
2,Debt Consolidation Loan,47416.443852
3,Home Equity Loan,49754.329214
4,Mortgage Loan,49604.188106


In [ ]:
fig = px.histogram(
    data,
    x='primary_loan_type',
    y='Annual_Income',
    color='primary_loan_type',
    title='Histplot for Loan type vs Annual Income '
)

fig.update_layout(
    title_font_size=20,
    title_font_weight = 'bold',
    font= dict(size=12,weight='bold'),
    yaxis_title='Annual income'
)

fig.show()


##### 1. Why did you pick the specific chart?

**Histogram** to see the **distribution of average annual income** wrt loan types. This is done to compare customers annual incomes and type of loans they have choosen.

##### 2. What is/are the insight(s) found from the chart?

1. **Accross all loan types** the **annual income level is between 48k-50k**. which suggests,**annual income level does not influence the loan types**.
2. **Data gap**(loan type is missing) for **high annual income cutomers(63k)**.

##### 3. Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

**POSITIVES:**
1. **Paisabazar is not over dependent on single income segment**.Loans and **products are accessible to customers/individuals with variuos annual income ranges**.

**Negatives:**
1. **Data issues**: **No Data** and **Not specified** categories have highers incomes but have any loan type/missing loan type data.

#### Chart - 4 BIVARIATE ANALYSIS (scatter plot for no.of cards,utilization ratio)

In [ ]:
pdf.head()

,ID,Customer_ID,Month,Name,Age,SSN,Occupation,Annual_Income,Monthly_Inhand_Salary,Num_Bank_Accounts,Num_Credit_Card,Interest_Rate,Num_of_Loan,Type_of_Loan,Delay_from_due_date,Num_of_Delayed_Payment,Changed_Credit_Limit,Num_Credit_Inquiries,Credit_Mix,Outstanding_Debt,Credit_Utilization_Ratio,Credit_History_Age,Payment_of_Min_Amount,Total_EMI_per_month,Amount_invested_monthly,Payment_Behaviour,Monthly_Balance,Credit_Score,primary_loan_type,credit_cat
0,5634,3392,1,Aaron Maashoh,23.0,821000265.0,Scientist,19114.12,1824.843333,3.0,4.0,3.0,4.0,"Auto Loan, Credit-Builder Loan, Personal Loan,...",3.0,7.0,11.27,4.0,Good,809.98,26.822620,265.0,No,49.574949,21.46538,High_spent_Small_value_payments,312.494089,Good,Auto Loan,2
1,5635,3392,2,Aaron Maashoh,23.0,821000265.0,Scientist,19114.12,1824.843333,3.0,4.0,3.0,4.0,"Auto Loan, Credit-Builder Loan, Personal Loan,...",3.0,4.0,11.27,4.0,Good,809.98,31.944960,266.0,No,49.574949,21.46538,Low_spent_Large_value_payments,284.629162,Good,Auto Loan,2
2,5636,3392,3,Aaron Maashoh,23.0,821000265.0,Scientist,19114.12,1824.843333,3.0,4.0,3.0,4.0,"Auto Loan, Credit-Builder Loan, Personal Loan,...",3.0,7.0,11.27,4.0,Good,809.98,28.609352,267.0,No,49.574949,21.46538,Low_spent_Medium_value_payments,331.209863,Good,Auto Loan,2
3,5637,3392,4,Aaron Maashoh,23.0,821000265.0,Scientist,19114.12,1824.843333,3.0,4.0,3.0,4.0,"Auto Loan, Credit-Builder Loan, Personal Loan,...",5.0,4.0,6.27,4.0,Good,809.98,31.377862,268.0,No,49.574949,21.46538,Low_spent_Small_value_payments,223.451310,Good,Auto Loan,2
4,5638,3392,5,Aaron Maashoh,23.0,821000265.0,Scientist,19114.12,1824.843333,3.0,4.0,3.0,4.0,"Auto Loan, Credit-Builder Loan, Personal Loan,...",6.0,4.0,11.27,4.0,Good,809.98,24.797347,269.0,No,49.574949,21.46538,High_spent_Medium_value_payments,341.489231,Good,Auto Loan,2


In [ ]:
##data preperation
data= pdf.groupby('Customer_ID').agg({'Num_Credit_Card':'mean',
                                      'Credit_Utilization_Ratio':'mean',
                                       'credit_cat':'last'}).reset_index()       ## takign most recent credit category/score

In [ ]:
data.info()   ## has null values

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 12500 entries, 0 to 12499
Data columns (total 4 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Customer_ID               12500 non-null  int64  
 1   Num_Credit_Card           12500 non-null  float64
 2   Credit_Utilization_Ratio  12500 non-null  float64
 3   credit_cat                12500 non-null  int64  
dtypes: float64(2), int64(2)
memory usage: 390.8 KB


In [ ]:
data['Num_Credit_Card'] = pd.to_numeric(data['Num_Credit_Card'], errors='coerce')
data['Credit_Utilization_Ratio'] = pd.to_numeric(data['Credit_Utilization_Ratio'], errors='coerce')       ##to avoid any kind of data inconsistencies
data['credit_cat'] = pd.to_numeric(data['credit_cat'], errors='coerce')



In [ ]:
data.info() ## has null values

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 12500 entries, 0 to 12499
Data columns (total 4 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Customer_ID               12500 non-null  int64  
 1   Num_Credit_Card           12500 non-null  float64
 2   Credit_Utilization_Ratio  12500 non-null  float64
 3   credit_cat                12500 non-null  int64  
dtypes: float64(2), int64(2)
memory usage: 390.8 KB


In [ ]:
# Chart - 4 scatter plot
fig = px.scatter(
    data,
    x='Num_Credit_Card',
    y='Credit_Utilization_Ratio',
    color='credit_cat',
    size_max=50,
    labels={
        'Num_Credit_Card': 'no of credit cards',
        'Credit_Utilization_Ratio':'credit(utilization) used per limit'

    },
    title='No.of Credit cards and the utilization ratio'
)

fig.update_traces(
    marker=dict(
        opacity=0.75,
        line=dict(width=1, color='DarkSlateGrey')
    )
)

fig.update_layout(
    title_font_size=20,
    title_font_weight='bold',
    xaxis=dict(
        showgrid=True,
        gridcolor='lightgrey'
    ),
    yaxis=dict(
        showgrid=True,
        gridcolor='lightgrey'
    ),
    legend_title_text='Region',
    height=500,
    template='plotly_white'
)

fig.show()


##### 1. Why did you pick the specific chart?

Scatter plot is used to identify relations between two contious variables. Here in this scatter plot we are trying to  **find the relation between credit utilization ratio and no.of credit card owned by the customer.**

##### 2. What is/are the insight(s) found from the chart?

1. Customers with **fewer credit cards(1-3)** maitain **Good credit score**. The customers who hold **more credit cards(6-10+)** have **Poor and standard credit scores.**

2. As the **credit cards number increases**, customers **credit scores changes from good-->standard-->poor.**
3. The **majority of customers with credit cards(3-6)** maintained **standard credit score**,very **few** of them **have poor** and **good credit scores**.
4. Credit **utilization ratio crosses 30%** irrespective of no.of credit cards. Majority of **customers with higher credit cards** have **try to keep the utilization ratio below 35%**. **Lower credit utilization ratio (below 25%)** is **observed** for some customers **irrespective of the no.of credit cards.**

##### 3. Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

**POSITIVES:**

1. **No.of Credit cards** is **negetively related** to the **credit score**. (Fewer credit cards better credit score).

2. **Best credit behaviour profile**:
   --> No.of Cards= 1-3 cards
   --> Credit utilization ratio = below 28%
   --> Good credit score

**NEGATIVES:**
1. **Accumilation of credit cards leads to credit dependence and degrading credit score** and **indicates low financial strength**.


#### Chart - 5 BIVARIATE ANALYSIS for no.of delayed payments and outstanding debt

In [ ]:
data = pdf.groupby('Customer_ID').agg({
    'Num_of_Delayed_Payment':'mean',
    'Outstanding_Debt':'mean',
    'Credit_Utilization_Ratio':'mean',
    'credit_cat':'last'}).reset_index()

In [ ]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 12500 entries, 0 to 12499
Data columns (total 5 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Customer_ID               12500 non-null  int64  
 1   Num_of_Delayed_Payment    12500 non-null  float64
 2   Outstanding_Debt          12500 non-null  float64
 3   Credit_Utilization_Ratio  12500 non-null  float64
 4   credit_cat                12500 non-null  int64  
dtypes: float64(3), int64(2)
memory usage: 488.4 KB


In [ ]:
data.sort_values(by='Outstanding_Debt',ascending=True,inplace=True,ignore_index=True)

In [ ]:
data.head()

,Customer_ID,Num_of_Delayed_Payment,Outstanding_Debt,Credit_Utilization_Ratio,credit_cat
0,1221,8.000,0.23,30.751913,2
1,14337,20.000,0.34,31.229410,1
2,50420,2.000,0.54,31.941398,0
3,6748,8.500,0.56,32.356600,1
4,43790,3.125,0.77,33.995784,1


In [ ]:
# Chart - 5 scatter plot
fig = px.scatter(
    data,
    x='Num_of_Delayed_Payment',
    y='Outstanding_Debt',
    color='credit_cat',
    trendline='ols',
    opacity=1,
    title='Outstanding Debt vs Delayed Payments'
)

fig.show()


##### 1. Why did you pick the specific chart?

**Scatter plot** is used to visualize the **relation between outstanding debts and no of delayed payments**, (which are two continous variables)



##### 2. What is/are the insight(s) found from the chart?

1. The **outstanding debt accumilates** as the **no.of delayed payments increases**.The upward tredline confirms **positive correlation**.

2. **Risk Segmentation**:

*   Low debt (0-1500)--> delayed payments < 7
*   Low to Moderate debt (0-2600)--> delayed payments < 15
*   Low to high debts (0-5000)--> delayed payments between (15-21)
*   Moderate to high debts (2600-5000)--> delay payments (21-25)

3. **Credit score segementation:**
*  Good score = debt below 1500
*  Standard score = Low to medium debt(below 2600)
*  Poor score = dominates medium to high debts(2600-5000)

4. **Low delays** = Compact cluster of debts
5. **High delays**= Wide dispersion of debts


##### 3. Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

**POSITIVES**:
1. **Delayed payments** can be used as the **Early risk indicators**,so that the bank can **flag the customers based in high debts**.

2. **Risk segmentation using credit scores and no.of delays**. so that the bank can **alert the customers** or **restructure the payments** or **freeze the credit limit** as **early** as possible.

NEGATIVES:
1. **Good and Standard** credit scores in **moderate and high debts**, suggesting **credit score alone can not explain debt accumilation**.

#### Chart - 6 BIVARIABLE ANALYSIS

---



In [ ]:
## data preperation
data = pdf[['Outstanding_Debt','Credit_History_Age','Payment_of_Min_Amount']]


In [ ]:
data.head()

,Outstanding_Debt,Credit_History_Age,Payment_of_Min_Amount
0,809.98,265.0,No
1,809.98,266.0,No
2,809.98,267.0,No
3,809.98,268.0,No
4,809.98,269.0,No


In [ ]:
# Chart - 6 Scatter plot
fig = px.scatter(
    data,
    x='Credit_History_Age',
    y='Outstanding_Debt',
    trendline='ols',
    opacity=0.5,)
fig.update_layout(
    title='Outstanding Debt vs credit history age',
    title_font_size=20,
    title_font_weight='bold',
    font = dict(size=12,weight='bold')
)

fig.show()

##### 1. Why did you pick the specific chart?

**Scatterplot** is generally used for continous-variabel analysis. Here scatterplot is used to analyse **how Outstanding debt is related to the credit history age**.


##### 2. What is/are the insight(s) found from the chart?

**Three strong vertical blocks**:
1. 0-60 months → very high outstanding debt (up to 5000)

2. 60-180 months → moderate debt to high debt (mostly 500-2500 dense, 2500-5000 spread slightly )

3. 180-400 months → low debt (mostly < 1500 ,dense cluster). As the credit score increases the outstanding debt decreases.

##### 3. Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

POSITIVES:
1. **Credit history age** can be used as a **risk indicator**:
*  lower age --> higher debt
*  higher age --> lower debt.

2. **Expereinced borrowers** try to controll their debt. These customers can be target for **pre-approved loans** and **premium credit cards**.

#### Chart - 7

> BIVARIATE ANALYSIS



In [ ]:
data = pdf.groupby('Customer_ID').agg({
    'Monthly_Inhand_Salary':'mean',
    'Amount_invested_monthly':'mean',
    'Payment_Behaviour':lambda x: x.mode()[0],
    'Credit_Score':lambda x: x.mode()
})

In [ ]:
data.head()

,Monthly_Inhand_Salary,Amount_invested_monthly,Payment_Behaviour,Credit_Score
Customer_ID,,,,
1006,1331.348333,45.301068,Low_spent_Small_value_payments,Poor
1007,1496.742500,30.373472,Low_spent_Small_value_payments,Standard
1008,2655.035833,50.930150,High_spent_Small_value_payments,Standard
1009,6692.636667,101.120201,High_spent_Medium_value_payments,Standard
1011,8433.546667,102.488034,Low_spent_Medium_value_payments,Standard


In [ ]:
data1 = data.groupby('Payment_Behaviour')['Monthly_Inhand_Salary'].median().sort_values(ascending=False).reset_index()

In [ ]:
data1

,Payment_Behaviour,Monthly_Inhand_Salary
0,High_spent_Large_value_payments,7097.453333
1,High_spent_Medium_value_payments,4194.567500
2,Low_spent_Medium_value_payments,3760.959583
3,High_spent_Small_value_payments,3031.296250
4,Low_spent_Large_value_payments,2691.350000
5,Low_spent_Small_value_payments,1523.987083


In [ ]:
# Chart - 7
fig = px.bar(
    data1,
    y='Monthly_Inhand_Salary',
    x='Payment_Behaviour',
    text_auto=False,
    title='Analysing payment behaviour of customers',
    color='Payment_Behaviour',
    opacity=1
)

fig.update_layout(
    title_font_size=20,
    title_font_weight='bold',
    font= dict(size=12,weight='bold'),
    bargap=0.2
)

fig.show()


##### 1. Why did you pick the specific chart?

**Bar chart** is used to **compare the payment behaviour** of the customers wrt their **monthly inhand salary**.



##### 2. What is/are the insight(s) found from the chart?

1. **High-spent-large-value payment behaviour** is observed in **csutomers** with **high monthly inhand salary(7097.45)**. This is followed by **High-spent-medium-value-payments behaviour customers** with monthly inhand **salary of** **(4194.57)**.There is a **significant drop** between these two payment behaviours.

2. **Low_spent_Medium_value_payments behaviour** is observed in customers with monthly inhand salary of **(3760.96)** and	**Low_spent_Large_value_payments** behaviour customers have a monthly inhand salary of **(2691.35)**. These customers are **lower salary**, but have **medium and large payments** suggesting these **customers are facing financial burden.**
3. **As the monthly inhand salary decreases:**
*  spending shifts from **High-->Low**
*  payment behaviour shifts from **High-->Medium-->Low**

##### 3. Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

**POSITIVES:**
1. **Strong alignment** between **capacity and payment behavior**. **High income and high payment behaviour customers** can be targeted for **Credit upgrade and premium loans.**

**NEGATIVES:**
1. **Low-income and large paymemts/medium payments behaviour customers** may **accumilate debt**, **reformed payment structure** might be required for them.

#### Chart - 8 MULTIVARIABLE ANALYSIS WITH SUBPLOTS

In [ ]:
data = pdf.groupby('Customer_ID').agg({
    'Monthly_Inhand_Salary':'mean',
    'Total_EMI_per_month':'mean',
    'Amount_invested_monthly':'mean',
    'Monthly_Balance':'mean',
    'Credit_Score':lambda x:x.mode()[0]

}).reset_index()

In [ ]:
data.head()

,Customer_ID,Monthly_Inhand_Salary,Total_EMI_per_month,Amount_invested_monthly,Monthly_Balance,Credit_Score
0,1006,1331.348333,27.442089,45.301068,314.626667,Poor
1,1007,1496.742500,45.745700,30.373472,285.607087,Standard
2,1008,2655.035833,0.000000,50.930150,362.407369,Standard
3,1009,6692.636667,188.456595,101.120201,523.000149,Standard
4,1011,8433.546667,257.738646,102.488034,445.787831,Standard


In [ ]:
# Chart - 8 visualization code
import plotly.graph_objects as go
from plotly.subplots import make_subplots

fig = make_subplots(
    rows=2, cols=2,
    subplot_titles=(
        'credit score wrt monthly salary',
        'credit score wrt monthly EMI',
        'credit score wrt monthly investments',
        'Credit score wrt monthly balance'
    )
)

# Box
fig.add_trace(
    go.Box(x=data['Credit_Score'],y=data['Monthly_Inhand_Salary'],name='monthly salary'),
    row=1, col=1)

# Box
fig.add_trace(
    go.Box(x=data['Credit_Score'],y=data['Total_EMI_per_month'], name='EMI monthly'),
    row=1, col=2)
# Box
fig.add_trace(
    go.Box(x=data['Credit_Score'],y=data['Amount_invested_monthly'], name='monthly investment'),
    row=2, col=1)
# Box
fig.add_trace(
    go.Box(x=data['Credit_Score'],y=data['Monthly_Balance'], name='monthly balance'),
    row=2, col=2)


fig.update_layout(
    title='FACTORS INFLUENCING CREDIT SCORE',
    title_font_size=20,
    title_font_weight = 'bold',
    font= dict(size=12,weight='bold')
)

fig.show()


##### 1. Why did you pick the specific chart?

**Box plot subplots** are used to analyse the **distribution of factors influencing credit scores.**



##### 2. What is/are the insight(s) found from the chart?

1. As the **credit score improves** from **Poor-->Standard-->Good**:
* **Median of monthly income increases**(39.72k-->45.76k-->56.23k).
*  The **spread** of the incomes also **widens** and the **oultiers decreases**.

2. Monthly **EMI overlaps heavily** accross **all credit scores**. Customers with **Good credit scores also pay high EMI** (**Many outliers** as the Cedit score improoves)

3. **Monthly investments:**
*  **High** for **Good credit score customers**.
*  **Moderate** for **Standard credit score customers**.
*  **Low investments** for **Poor credit score customers**.
*  **Accross credit scores** the there are **few high monthly investments**(outliers)

4. **Monthtly Balance:**
*  **High** for **Good credit score customers**(few outliers).
*  **Moderate** for **Standard credit score customers**(significant no.of outliers)
*  **Low** for **Poor credit score customers**(many outliers/high monthly balance)

##### 3. Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

**POSITIVES:**
1. All these factors can be used as the **KPIs** for:
*  **Assessing Risk**.
*  **Credit policy design**.
*  **financial behaviour modeling**.

#### Chart - 9 BIVARIABLE ANALYSIS

In [ ]:
data = pdf.groupby('Customer_ID').agg({
    'Outstanding_Debt':'mean',
    'Payment_of_Min_Amount':lambda x:x.mode()[0],
    'Payment_Behaviour':lambda x:x.mode()[0]
}).reset_index()

In [ ]:
data.head()

,Customer_ID,Outstanding_Debt,Payment_of_Min_Amount,Payment_Behaviour
0,1006,1941.73,Yes,Low_spent_Small_value_payments
1,1007,993.15,No,Low_spent_Small_value_payments
2,1008,1138.97,Yes,High_spent_Small_value_payments
3,1009,982.44,Yes,High_spent_Medium_value_payments
4,1011,1371.80,Yes,Low_spent_Medium_value_payments


In [ ]:
# Chart - 9 Histplot code
fig = px.histogram(
    data,
    x='Payment_Behaviour',
    y='Outstanding_Debt',
    color='Payment_of_Min_Amount',
    title='Histplot for Payment Behaviour vs Debt'
)

fig.update_layout(
    title_font_size=20,
    title_font_weight = 'bold',
    font= dict(size=12,weight='bold'),
    yaxis_title='Due amount'
)

fig.show()

##### 1. Why did you pick the specific chart?

Histogram to see **distribution of outstanding debt** wrt to payment behaviour and payment of min amount.

##### 2. What is/are the insight(s) found from the chart?

1. **Low-spent-small-value-payment** behaviour customers have **high outstanding debt(7M)**. **Majority** of these customers **pay minimum amount**.

2. **High-spent-medium-value-payment** and **High-spent-large-value-payment** behaviour customers have **moderate outstanding debts**, most of them pay minimum amount and **significant no.of customers do not minimum amount**.
3. **Low-spent-Large-value-payment** behaviour customers have **low outstanding debt.**

##### 3. Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

1. **High spending is not risky if repayment behaviour is strong**, **paying only minimum amount will lead to debt accumilation.**


#### Chart - 10 BIVARIATE ANALYSIS

In [ ]:
data = pdf.groupby('Customer_ID').agg({'Age':'median',
                                       'Num_of_Loan':'median',
                                       'Annual_Income':'mean'

}).reset_index()

In [ ]:
data.head()

,Customer_ID,Age,Num_of_Loan,Annual_Income
0,1006,38.0,2.0,16756.18
1,1007,48.0,3.0,21212.91
2,1008,37.0,0.0,33540.43
3,1009,22.0,4.0,80983.64
4,1011,44.0,3.0,104142.56


In [ ]:
# Chart - 10 visualization code

fig = make_subplots(
    rows=1, cols=2,
    subplot_titles=(
        'Boxplot for no.of loans vs annual income',
        'Scatterplot for age vs annual income'
    )
)
fig.add_trace(
    go.Scatter(x=data['Age'],y=data['Annual_Income'],name='Age vs Income',mode='markers'),
    row=1, col=2)

fig.add_trace(
    go.Box(x=data['Num_of_Loan'],y=data['Annual_Income'],name='Loans vs Income'),
    row=1, col=1)


fig.update_layout(
    title='AGE VS LOANS VS INCOME RELATIONSHIPS',
    title_font_size=20,
    title_font_weight = 'bold',
    font= dict(size=12,weight='bold'),
    width=1000
)

fig.show()


##### 1. Why did you pick the specific chart?

Subplots for multivariable analysis. These **subplots are used to analyse multiple variables at a time using different charts**.

##### 2. What is/are the insight(s) found from the chart?

**ANNUAL INCOME VS LOANS:**
*  Customers with **(0-2 loans)** have **highest medain income** (wide spread).
*  Customers with **(2-4 loans)** have **Moderate median income** (medium spread,few outliers).
*  Customers with **(5-8+ loans)** have **Low median income**.

**AGE VS ANNUAL INCOME:**
*  Accross age groups customers have wide range/variablility of annual incomes. Customers with age **above 20 have more higher incomes.**

##### 3. Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

None

#### Chart - 11 UNIVARIATE ANALYSIS

In [ ]:
data = pdf.groupby('Customer_ID').agg({'Age':'median','Credit_Score':lambda x:x.mode()[0]}).reset_index()

In [ ]:
data.head()

,Customer_ID,Age,Credit_Score
0,1006,38.0,Poor
1,1007,48.0,Standard
2,1008,37.0,Standard
3,1009,22.0,Standard
4,1011,44.0,Standard


In [ ]:
# Chart - 11 visualization code

fig = px.box(
    data,
    y='Age',
    x='Credit_Score',
    color='Credit_Score',
    title='Box plot for Credit score vs Age'
)

fig.update_layout(
    title_font_size=20,
    title_font_weight = 'bold',
    font= dict(size=12,weight='bold'),
    yaxis_title='Age'
)

fig.show()

##### 1. Why did you pick the specific chart?

**Box chart** to see **age distribution accross different credit scores.**



##### 2. What is/are the insight(s) found from the chart?

1. Customers with Good credit scores are older(median=36, upper=46,lower=27)
2. Customers with Standard credit scores are middle aged (median=33, upper=42)lower=25)
3. Customers with Poor credit scores have median age =31, upper age =46,lower=27)


##### 3. Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

Answer Here

#### Chart - 12 BIVARIATE ANALYSIS

In [ ]:
data = pdf.groupby('Customer_ID').agg({'Credit_Score':lambda x:x.mode()[0],
                                       'Total_EMI_per_month':'mean',
                                       'Payment_of_Min_Amount':lambda x:x.mode()[0]

}).reset_index()

In [ ]:
data.head()

,Customer_ID,Credit_Score,Total_EMI_per_month,Payment_of_Min_Amount
0,1006,Poor,27.442089,Yes
1,1007,Standard,45.745700,No
2,1008,Standard,0.000000,Yes
3,1009,Standard,188.456595,Yes
4,1011,Standard,257.738646,Yes


In [ ]:
data = data.groupby(['Credit_Score','Payment_of_Min_Amount'])['Total_EMI_per_month'].mean().reset_index()

In [ ]:
data.head()

,Credit_Score,Payment_of_Min_Amount,Total_EMI_per_month
0,Good,NM,150.693708
1,Good,No,109.874498
2,Good,Yes,186.425489
3,Poor,NM,97.086884
4,Poor,No,88.645371


In [ ]:
# Chart - 12
fig = px.bar(
    data,
    x='Credit_Score',
    y='Total_EMI_per_month',
    color='Payment_of_Min_Amount',
    title='Average Monthly EMI by Credit Score and Minimum Payment Behaviour',
    labels={
        'Total_EMI_per_month': 'Average Monthly EMI',
        'Credit_Score': 'Credit Score Category',
        'Payment_of_Min_Amount': 'Pays Only Minimum?'
    }
)

fig.show()


##### 1. Why did you pick the specific chart?

**Bar chart to** **compare credit scores wrt Avg.mothly EMI and minimum payment category**
[link text](https://)


##### 2. What is/are the insight(s) found from the chart?

1. **Good credit customer**s have **high average.monthly EMI(450)**, followed by **Poor credit score customers with Monthly EMI of (300)**, **Standard credit score customers** have **low avg.monthly EMI of (250+)**.

2. **Accross all the credit scores** there are **significant no.of people** who are **paying more than minimum required payment** and there is **data of customers who minimum payment category is not mentioned/missing data (intentionally)**

##### 3. Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

**POSITIVES:**
1. Offer **higher credit limit to the low-risk customers**(Good-credit score, pays more than minimum amount),they have **better repayment capacity**.

**NEGATIVES**:
**NM category limts the full analysis.**

#### Chart - 13 MULTIVARIABLE ANALYSIS

In [ ]:
data = pdf.groupby('Customer_ID').agg({
    'Age':'median',
    'Annual_Income':'mean',
    'Monthly_Inhand_Salary':'mean',
    'Num_Bank_Accounts':'median',
    'Num_Credit_Card':'median',
    'Num_of_Loan':'median',
    'Num_of_Delayed_Payment':'median',
    'Changed_Credit_Limit':'mean',
    'Num_Credit_Inquiries':'median',
    'Outstanding_Debt':'mean',
    'Credit_Utilization_Ratio':'mean',
    'Credit_History_Age':'median',
    'Total_EMI_per_month':'mean',
    'Amount_invested_monthly':'mean',
    'Monthly_Balance':'mean',
    'credit_cat':'last'

}).reset_index()

In [ ]:
data.head()

,Customer_ID,Age,Annual_Income,Monthly_Inhand_Salary,Num_Bank_Accounts,Num_Credit_Card,Num_of_Loan,Num_of_Delayed_Payment,Changed_Credit_Limit,Num_Credit_Inquiries,Outstanding_Debt,Credit_Utilization_Ratio,Credit_History_Age,Total_EMI_per_month,Amount_invested_monthly,Monthly_Balance,credit_cat
0,1006,38.0,16756.18,1331.348333,9.0,6.0,2.0,12.0,10.660,8.0,1941.73,31.727384,185.5,27.442089,45.301068,314.626667,0
1,1007,48.0,21212.91,1496.742500,3.0,4.0,3.0,19.0,5.630,1.0,993.15,36.313739,349.5,45.745700,30.373472,285.607087,0
2,1008,37.0,33540.43,2655.035833,6.0,3.0,0.0,11.0,14.110,10.0,1138.97,28.688272,295.5,0.000000,50.930150,362.407369,1
3,1009,22.0,80983.64,6692.636667,6.0,3.0,4.0,18.0,17.285,7.0,982.44,34.677793,334.5,188.456595,101.120201,523.000149,1
4,1011,44.0,104142.56,8433.546667,3.0,5.0,3.0,14.0,15.780,6.0,1371.80,33.611631,182.5,257.738646,102.488034,445.787831,1


In [ ]:
corr_data = data[['Age',
    'Monthly_Inhand_Salary',
    'Num_Bank_Accounts',
    'Num_Credit_Card',
    'Num_of_Loan',
    'Num_of_Delayed_Payment',
    'credit_cat']].corr()

corr_data1 = data[['Changed_Credit_Limit',
    'Num_Credit_Inquiries',
    'Outstanding_Debt',
    'Credit_Utilization_Ratio',
    'Credit_History_Age',
    'Total_EMI_per_month',
    'Amount_invested_monthly',
    'Monthly_Balance',
    'credit_cat']].corr()

In [ ]:
# Chart - 13 visualization code
fig = px.imshow(
    corr_data,
    text_auto='.2f',
    color_continuous_scale='cividis',
    title='Correlation Heatmap',
    zmin=-1,
    zmax=1,
    aspect='auto'
)
fig.update_traces(
    textfont=dict(
        size=13,
        color='black'
    ))

fig.update_layout(
    font=dict(size=22, family='Arial Black'),
    width=1200,
    height=600,
    margin=dict(l=120, r=80, t=100, b=80)
)

fig.show()

##### 1. Why did you pick the specific chart?

**Correlation heatmap** to check the **Correlation between variables at a time**.

*  -1--> Strong negative correlation.
*   0--> No correlation.
*  +1--> Strong positive correlation.

##### 2. What is/are the insight(s) found from the chart?

1. **Correlation with Credit category/Credit score:**
*  **Age is positvely correlated** with credit score, but the strength of correlation is **weak (0.14)**.
*  **Monthly salary** is **postively correlated**,but **weak strength(0.19)**.
*  **No.of Bank accounts** is **Negatively correlated**, with **moderate strength(-0.34)**.
*  **No.of loans,No.of credit cards and No.of delayed payments** are all **Negatively correlated** with credit score, each with **moderate strength (-0.36,-0.32,-0.32)** respectively.

2. **Correlation with Delayed payments:**
*  **Age is negatively correlated**, with **weak strength of (-0.19)**.
*  **Monthly salary is negatively correlated**, with a **weak correlation strength of (-0.29)**.
*  **No.of loans,No.of credit cards and No.of delayed payments** are all **Positively correlated** with no.of delayed payments, each with **high moderate strength(0.61),moderate strength(0.43),and moderate strength(0.48)** respectively.

3. **Correlation with no.of loans**:
* **Age and monthly salary are negatively correlated** with no.of loans, with a **weak strength** of **(-0.21) and (-0.25)** respectively.

*  **No.of bank accounts** and **no.of credit cards** are **positively correlated** with no.of loans, with a **Moderate strength** of **(0.47)** and **(0.42)** respectively.

4 **Correlation with no.of credit cards**:
*  **Age** and **montlhy salary** are **negatively correlated** with the no.of credit cards, with a correlation strength of **(-0.15) and (-0.22) weak**.

5. **Age** and **monthly salary** are **negatively correlated** with **no.of bank acoounts** but has **weak strength (-0.19 and -0.28)**.

6. There is **no correlation/very low correlation with monthly salary and age.**

##### 3. Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

**POSITIVES:**
1. **Credit Overexposure**, customers with **More financial products have higher repayment delays.** (Identify them)
2. **Credit score** is a **strong risk indicator** (Better score-->few loans-->few cards-->less delays).
3. **Credit risk**/Loan risk is **not age driven**.

**NEGATIVES:**
1. **Aggressive lending** could lead to defaults. **Income only criteria** for loan approoval should be avoided. **More card-->More bank accounts--> No loyalty**.

#### Chart - 14 - Correlation Heatmap

In [ ]:
# Correlation Heatmap visualization code
fig = px.imshow(
    corr_data1,
    text_auto='.2f',
    color_continuous_scale='cividis',
    title='Correlation Heatmap',
    zmin=-1,
    zmax=1,
    aspect='auto'
)
fig.update_traces(
    textfont=dict(
        size=13,
        color='black'
    ))

fig.update_layout(
    font=dict(size=22, family='Arial Black'),
    width=1200,
    height=600,
    margin=dict(l=120, r=80, t=100, b=80)
)

fig.show()

**1. Credit score correlation:**
*  Credit score <--> Changed_Credit_Limit (-0.15 weak,negative)
*  Credit score <--> Num_Credit_Inquiries (-0.39 moderate,negaitve)
*  Credit score <--> Outstanding_Debt (-0.35 moderate,negative)
*  Credit score <--> Credit_Utilization_Ratio (0.12 weak,positive)
*  Credit score <--> Credit_History_Age (0.35 moderate,positive)
*  Credit score <--> Total_EMI_per_month (0.03,No correlation)
*  Credit score <--> Amount_invested_monthly (0.15, weak,positive)
*  Credit score <--> Monthly_Balance (0.26 weak,positive)

**2. Monthly Blanace correlation**:

*  Monthly balance <--> Changed_Credit_Limit (-0.26 weak,negative)
*  Monthly balance <--> Num_Credit_Inquiries (-0.40 moderate,negative)
*  Monthly balance <--> Outstanding_Debt  (-0.41 moderate,negative)
*  Monthly balance <--> Credit_Utilization_Ratio (0.47 moderate,positive)
*  Monthly balance <--> Credit_History_Age (0.41 moderate, positive)
*  Monthly balance <--> Total_EMI_per_month (0.10 very weak,positive)
*  Monthly balance <--> Amount_invested_monthly (0.61 high moderate,positive)

**3.Correlation with Amount invested monthly:**
*  Amount invested monthly <--> Changed_Credit_Limit (-0.15 weak,negative)
*  Amount invested monthly <--> Num_Credit_Inquiries (-0.22 weak,negative)
*  Amount invested monthly <--> Outstanding_Debt  (-0.22 weak,negative)
*  Amount invested monthly <--> Credit_Utilization_Ratio (0.34 moderate,positive)
*  Amount invested monthly <--> Credit_History_Age (0.23 moderate, positive)
*  Amount invested monthly <--> Total_EMI_per_month (0.39 moderate, positive)

4. **Total EMI per month** is **negatively correlated** with **credit-history-age (weak strength)**, **positively correlated** with **outsanding debt(0.12,weak strength)** and  **no.of credit inquires(0.10, weak strength).**

**5.Credit history age correlation:**
*  Credit history age <--> Changed_Credit_Limit(-0.44 moderate, Negative)
*  Credit history age <--> Num_Credit_Inquiries(-0.62 high moderate,Negative)
*  Credit history age <--> Outstanding_Debt (-0.63 high moderate,Negative)
*  Credit history age <--> Credit_Utilization_Ratio (0.18 weak,positive)

**6.credit utilization ratio:**
*  credit utilization ratio <--> Changed_Credit_Limit(-0.13 weak, Negative)
*  credit utilization ratio <--> Num_Credit_Inquiries(-0.20 weak,Negative)
*  credit utilization ratio <--> Outstanding_Debt (-0.18 weak,Negative)

**7. Outstanding debt**
* **Postively correlated** with Changed_Credit_Limit(0.48,moderate strength) and Num_Credit_Inquiries(0.60 high moderate strength)

 **no of credit enquires is positively correletd with changed credit card limit with moderate strength of (0.40)**

##### 2. What is/are the insight(s) found from the chart?

**POSITIVES**:
1. **High engageement** by customers who request credit limit change.
2. Customers with **high monthly investments** can targeted for **Wealth products, stock investments etc.**
3. **Higher credit history age stabilises** loans,debts,scores, and increase montlhy balances, investments. These **experienced customers** can be targeted for **pre-approoved loans,premium credit cards**.

4. **Monthly EMI** is **not a big risk**. **Affordable EMIs** for all customers.

**NEGATIVES:**
1. **Frequent credit checks and higher outstanding debts** damages credit score and credit history age.
2. **Lower montlhy balance** could be a warning signal,Could lead to **increased collection and recovery costs ** due to missed payments.

## **5. Solution to Business Objective**

#### What do you suggest the client to achieve Business Objective ?
Explain Briefly.

1. **Income + good behaviorial scorecard** should be used for the **loan approval**,also reduces risk.
2. Implement **real time credit utilization ratio alert**  for early risk detection.
*  Warning-->30-35%
*  Restriction--> above 40%
3. **Offer credit limit increase** instead of more no.of cards to **low-risk customers** for **loyality** and **reduced collection/recovery costs**.
4. **KPIs for:**

**-->**  **Financial behaviour:**
*  Credit utilization ratio
*  outstanding debt
*  credit history age,payment behaviour and delayed payments.

**-->**  **Customer engagement**:
*  credit limit change.
*  monhtly EMI
*  High credit history age

5. **-->**  **Credit policy design:**
*  Good score,high history,low utilization--> pre-aprooved loans,premium cards
*  standard scores and moderate utilization --> Credit factors education, behavioural change.
*  High utilization and delays --> restrict limit, restructure payments.

6. **-->**  **Risk for Buisiness:**
*  Aggressive lending-->credit overexposure.
*  frequent credit enquiry --> degrades score.
*  Lower monthly balance
*  NM/ data gap in payment behaviour





# **CONCLUSION**:
**Credit risk is highly driven by the customer behaviour than income, Smart utilization monitoring, Credit education of standard score customers, early intervention reduces defaults and improoves customer life-time value**